crates/typst/src/math/underover.rs

```diff
@@ -267,29 +403,29 @@ fn layout_underoverspreader(
     let stretched = glyph.stretch_horizontal(ctx, body.width(), Abs::zero());
 
     let mut rows = vec![MathRun::new(vec![body]), stretched.into()];
+    if let Some(annotation) = annotation {
+        let (above_style, below_style);
+        let annot_styles = match position {
+            Position::Above => {
+                above_style = style_for_subscript(styles);
+                styles.chain(&above_style)
+            }
+            Position::Below => {
+                below_style = style_for_superscript(styles);
+                styles.chain(&below_style)
+            }
+        };
+        rows.push(ctx.layout_into_run(annotation, annot_styles)?);
+    }
 
-    let (sup_style, sub_style);
-    let row_styles = if reverse {
-        sup_style = style_for_subscript(styles);
-        styles.chain(&sup_style)
-    } else {
-        sub_style = style_for_subscript(styles);
-        styles.chain(&sub_style)
+    let baseline = match position {
```

Comment:
I think, at the cost of duplicating the `ctx.layout_into_run(annotation, annot_styles)?`, we could put everything into one match and simplify things a bit. no need to do the `let (above_style, below_style);` lifetime dance and we can use `insert` directly rather than `reverse`.